src/main/java/com/google/api/codegen/configgen/ConfigYamlNodeReader.java

```diff
@@ -0,0 +1,249 @@
+/* Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.configgen;
+
+import com.google.api.codegen.configgen.nodes.ConfigNode;
+import com.google.api.codegen.configgen.nodes.FieldConfigNode;
+import com.google.api.codegen.configgen.nodes.ListItemConfigNode;
+import com.google.api.codegen.configgen.nodes.NullConfigNode;
+import com.google.api.codegen.configgen.nodes.ScalarConfigNode;
+import com.google.common.base.Joiner;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import java.util.List;
+import org.yaml.snakeyaml.nodes.MappingNode;
+import org.yaml.snakeyaml.nodes.Node;
+import org.yaml.snakeyaml.nodes.NodeTuple;
+import org.yaml.snakeyaml.nodes.ScalarNode;
+import org.yaml.snakeyaml.nodes.SequenceNode;
+
+public class ConfigYamlNodeReader {
+  private static final String TYPE_KEY = "type";
+
+  private final List<String> lines;
+  private final ConfigHelper helper;
+
+  public ConfigYamlNodeReader(List<String> lines, ConfigHelper helper) {
+    this.lines = lines;
+    this.helper = helper;
+  }
+
+  public ConfigNode readMessageNode(int prevLine, MappingNode node, Descriptor messageType) {
```

Comment:
Comments please. (What do the `node` and `messageType` represent? What is contained in the `ConfigNode` that they're read into?)